ray-operator/controllers/ray/common/pod.go

```diff
@@ -176,7 +265,45 @@ func BuildPod(podTemplateSpec v1.PodTemplateSpec, rayNodeType rayiov1alpha1.RayN
 		setInitContainerEnvVars(&pod.Spec.InitContainers[index], svcName)
 	}
 
-	setContainerEnvVars(&pod.Spec.Containers[rayContainerIndex], rayNodeType, rayStartParams, svcName)
+	setContainerEnvVars(&pod, rayContainerIndex, rayNodeType, rayStartParams, svcName)
+
+	// health check only if HA enabled
+	if podTemplateSpec.Annotations != nil {
+		if enabledString, ok := podTemplateSpec.Annotations[RayHAEnabledAnnotationKey]; ok {
+			if strings.ToLower(enabledString) == "true" {
+				// Ray HA is enabled and we need to add health checks
+				if pod.Spec.Containers[rayContainerIndex].ReadinessProbe == nil {
+					// it is possible that some user have the probe parameters to override the default,
+					// in this case, this if condition is skipped
+					probe := &v1.Probe{
+						InitialDelaySeconds: DefaultReadinessProbeInitialDelaySeconds,
+						TimeoutSeconds:      DefaultReadinessProbeTimeoutSeconds,
+						PeriodSeconds:       DefaultReadinessProbePeriodSeconds,
+						SuccessThreshold:    DefaultReadinessProbeSuccessThreshold,
+						FailureThreshold:    DefaultReadinessProbeFailureThreshold,
+					}
+					pod.Spec.Containers[rayContainerIndex].ReadinessProbe = probe
+				}
+				// add readiness probe exec command in case missing.
+				initReadinessProbeHandler(pod.Spec.Containers[rayContainerIndex].ReadinessProbe, rayNodeType)
+
+				if pod.Spec.Containers[rayContainerIndex].LivenessProbe == nil {
+					// it is possible that some user have the probe parameters to override the default,
+					// in this case, this if condition is skipped
+					probe := &v1.Probe{
+						InitialDelaySeconds: DefaultLivenessProbeInitialDelaySeconds,
+						TimeoutSeconds:      DefaultLivenessProbeTimeoutSeconds,
+						PeriodSeconds:       DefaultLivenessProbePeriodSeconds,
+						SuccessThreshold:    DefaultLivenessProbeSuccessThreshold,
+						FailureThreshold:    DefaultLivenessProbeFailureThreshold,
+					}
+					pod.Spec.Containers[rayContainerIndex].LivenessProbe = probe
+				}
+				// add liveness probe exec command in case missing
+				initLivenessProbeHandler(pod.Spec.Containers[rayContainerIndex].LivenessProbe, rayNodeType)
```

Comment:
Just a thought for, maybe for later -- I wonder if these probes are useful for restarting pods even if HA isn't enabled.

Reply:
sounds good. we can have some discussion on this later.